qsp_transfer.py

```diff
@@ -0,0 +1,108 @@
+import os
+import sys
+import traceback
+import json
+import boto3
+import base64
+import time
+import re
+import logging
+
+from ftplib import FTP
+from datetime import datetime, timedelta
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def get_ftp_credentials():
+    client = boto3.client('secretsmanager')
+    get_secret_value_response = client.get_secret_value(SecretId=os.environ['SECRET_NAME'])
+
+    if 'SecretString' in get_secret_value_response:
+        secret = get_secret_value_response['SecretString']
+    else:
+        secret = base64.b64decode(get_secret_value_response['SecretBinary'])
+
+    return json.loads(secret)
+
+def create_data_file(date):
+    if not re.match(r"\d{2}-\d{2}-\d{4}", date):
+        raise Exception(f"Date parameter value '{date}' does not match expected format DD-MM-YYYY")
+
+    log_group = os.environ['LOG_GROUP_NAME']
+    start_time = int(time.mktime(time.strptime(f"{date} 00:00:00", '%d-%m-%Y %H:%M:%S'))) * 1000
```

Comment:
This feels like it could be pulled out into a separate function (`calculate_epoch_time`?) and refactored.